src/kvstore.h

```diff
@@ -4,6 +4,20 @@
 #include "dict.h"
 #include "adlist.h"
 
+/* maximum number of bins of keysizes histogram */
+#define MAX_KEYSIZES_BINS 48
+#define MAX_KEYSIZES_TYPES 5 /* static_assert at db.c verifies == OBJ_TYPE_BASIC_MAX */
+
+/* Per dict metadata. Memset zero on init. Managed outside kvstore */
+typedef struct {
+    uint64_t keysizes_hist[MAX_KEYSIZES_TYPES][MAX_KEYSIZES_BINS]; 
+} kvstoreDictMetadata;
+
+/* kvstore metadata. Memset zero on init. Managed outside kvstore */
+typedef struct {
+    uint64_t keysizes_hist[MAX_KEYSIZES_TYPES][MAX_KEYSIZES_BINS];
+} kvstoreMetadata;
```

Comment:
I wanted some clear declaration of this struct. 
One optionally attached to kvstore and another optionally per dict in kvstore.
I think the logic to update them is rather simple and clear. It  is being made via single function `updateKeysizesHist()`. 
So i didn't feel there is a need to further arrange this code.
lmk if you think otherwise.